### PR TITLE
UIIN-403: Use documented react-intl patterns instead of stripes.intl

### PR DESCRIPTION
**Pull request by anastasiaHahanova**
## Purpose
According to [UIIN-403](https://issues.folio.org/browse/UIIN-403), retrieving the intl object from this.context.intl or the stripes god object (stripes.intl) is a deprecated pattern.

This repo (edit-folder) needs to only use react-intl constructs outlined at https://github.com/folio-org/stripes-core/blob/master/doc/i18n.md

Use `<FormattedMessage>`, `<FormattedDate>`, and `<FormattedTime>`. The functions-only react-intl API  - formatMessage(), formatDate(), and formatTime() (accompanied by injectIntl) - shouldn't be needed.

React Intl docs: https://github.com/yahoo/react-intl/wiki

## Approach
- refactor formatMessage() to `<FormattedMessage>` where is possible
- add translation for hardcoded strings
- replace title with aria-label
- use [IntlConsumer](https://github.com/folio-org/stripes-core/pull/535) to avoid injectIntl HOC

DO NOT MERGE before [STCOM-535](https://github.com/folio-org/stripes-core/pull/535)

### CODE REVIEW

File: src/ViewHoldingsRecord.js [hunk 1]
```diff
@@ -229,19 +249,21 @@ class ViewHoldingsRecord extends React.Component {
...
-        />
+        <FormattedMessage id="ui-inventory.editHoldings">
+          {title => (
+            <IconButton
+              icon="edit"
+              id="clickable-edit-holdingsrecord"
+              style={{ visibility: !holdingsRecord ? 'hidden' : 'visible' }}
+              href={this.craftLayerUrl('editHoldingsRecord')}
+              onClick={this.onClickEditHoldingsRecord}
+              title={title}
```
Comment (cherewaty) [inline src/ViewHoldingsRecord.js]: We shouldn't be using `title` attributes anymore: https://issues.folio.org/browse/STCOM-296
`ariaLabel` should work fine here instead.

File: src/ViewHoldingsRecord.js [hunk 2]
```diff
@@ -299,69 +338,88 @@ class ViewHoldingsRecord extends React.Component {
...
+                      {ariaLabel => (
+                        <MultiColumnList
+                          id="list-statistical-codes"
+                          contentData={holdingsRecord.statisticalCodeIds.map((id) => { return { 'codeId': id }; })}
+                          visibleColumns={['Statistical code type', 'Statistical code']}
+                          formatter={{
+                            'Statistical code type':
+                              x => this.refLookup(referenceTables.statisticalCodeTypes,
+                                this.refLookup(referenceTables.statisticalCodes, _.get(x, ['codeId'])).statisticalCodeTypeId).name,
+                            'Statistical code':
```
Comment (Godlevskyi) [inline src/ViewHoldingsRecord.js]: Please add translations for the colums with the columnMapping prop

File: src/ViewHoldingsRecord.js [hunk 3]
```diff
@@ -370,121 +428,170 @@ class ViewHoldingsRecord extends React.Component {
...
-                    containerRef={(ref) => { this.resultsList = ref; }}
-                  />
+                <KeyValue
+                  label={<FormattedMessage id="ui-inventory.numberOfItems" />}
+                  value={_.get(holdingsRecord, ['numberOfItems'], [])}
+                />
+                {(holdingsRecord.holdingsStatements.length > 0) &&
+                  <FormattedMessage id="ui-inventory.holdingsStatements">
+                    {ariaLabel => (
+                      <MultiColumnList
```
Comment (Godlevskyi) [inline src/ViewHoldingsRecord.js]: Same here, no translations for the columns

File: src/ViewHoldingsRecord.js [hunk 4]
```diff
@@ -370,121 +428,170 @@ class ViewHoldingsRecord extends React.Component {
...
-                      'Holdings statement for supplements': x => _.get(x, ['statement']) || '',
-                      'Holdings statement for supplements note': x => _.get(x, ['note']) || '',
-                    }}
-                    ariaLabel={formatMsg({ id: 'ui-inventory.holdingsStatementForSupplements' })}
-                    containerRef={(ref) => { this.resultsList = ref; }}
-                  />
+                {(holdingsRecord.holdingsStatementsForSupplements.length > 0) &&
+                  <FormattedMessage id="ui-inventory.holdingsStatementForSupplements">
+                    {ariaLabel => (
+                      <MultiColumnList
```
Comment (Godlevskyi) [inline src/ViewHoldingsRecord.js]: Same here

File: src/ViewHoldingsRecord.js [hunk 5]
```diff
@@ -370,121 +428,170 @@ class ViewHoldingsRecord extends React.Component {
...
-                      'Holdings statement for indexes': x => _.get(x, ['statement']) || '',
-                      'Holdings statement for indexes note': x => _.get(x, ['note']) || '',
-                    }}
-                    ariaLabel={formatMsg({ id: 'ui-inventory.holdingsStatementForIndexes' })}
-                    containerRef={(ref) => { this.resultsList = ref; }}
-                  />
+                {(holdingsRecord.holdingsStatementsForIndexes.length > 0) &&
+                  <FormattedMessage id="ui-inventory.holdingsStatementForIndexes">
+                    {ariaLabel => (
+                      <MultiColumnList
```
Comment (Godlevskyi) [inline src/ViewHoldingsRecord.js]: Same here

File: src/ViewHoldingsRecord.js [hunk 6]
```diff
@@ -494,102 +601,117 @@ class ViewHoldingsRecord extends React.Component {
...
-                      'Link text': x => _.get(x, ['linkText']) || '',
-                      'Materials specified': x => _.get(x, ['materialsSpecification']) || '',
-                      'URL public note': x => _.get(x, ['publicNote']) || '',
-                    }}
-                    ariaLabel="Electronic access"
-                    containerRef={(ref) => { this.resultsList = ref; }}
-                  />
+                  <FormattedMessage id="ui-inventory.electronicAccess">
+                    {ariaLabel => (
+                      <MultiColumnList
```
Comment (Godlevskyi) [inline src/ViewHoldingsRecord.js]: Same here

File: src/ViewHoldingsRecord.js [hunk 7]
```diff
@@ -494,102 +601,117 @@ class ViewHoldingsRecord extends React.Component {
...
-                    formatter={{
-                      'Enumeration': x => x.enumeration,
-                      'Chronology': x => x.chronology,
-                    }}
-                    ariaLabel="Receiving history"
-                    containerRef={(ref) => { this.resultsList = ref; }}
-                  />
+                  <FormattedMessage id="ui-inventory.receivingHistory">
+                    {ariaLabel => (
+                      <MultiColumnList
```
Comment (Godlevskyi) [inline src/ViewHoldingsRecord.js]: Same here

File: src/ViewInstance.js [hunk 8]
```diff
@@ -414,90 +424,104 @@ class ViewInstance extends React.Component {
...
-                      this.refLookup(referenceTables.statisticalCodes, _.get(x, ['codeId'])).statisticalCodeTypeId).name,
-                  'Statistical code':
-                    x => this.refLookup(referenceTables.statisticalCodes, _.get(x, ['codeId'])).name,
-                }}
-                ariaLabel="Statistical codes"
-                containerRef={(ref) => { this.resultsList = ref; }}
-              />
+              <FormattedMessage id="ui-inventory.statisticalCodes">
+                {ariaLabel => (
+                  <MultiColumnList
```
Comment (Godlevskyi) [inline src/ViewInstance.js]: Same, no translations for thew columns

File: src/ViewInstance.js [hunk 9]
```diff
@@ -506,37 +530,47 @@ class ViewInstance extends React.Component {
...
-                contentData={instance.alternativeTitles}
-                rowMetadata={['alternativeTitleTypeId']}
-                visibleColumns={['Alternative title type', 'Alternative title']}
-                formatter={alternativeTitlesRowFormatter}
-                ariaLabel="Alternative titles"
-                containerRef={(ref) => { this.resultsList = ref; }}
-              />
+              <FormattedMessage id="ui-inventory.alternativeTitles">
+                {ariaLabel => (
+                  <MultiColumnList
```
Comment (Godlevskyi) [inline src/ViewInstance.js]: Same here

File: src/ViewInstance.js [hunk 10]
```diff
@@ -549,19 +583,23 @@ class ViewInstance extends React.Component {
...
-                contentData={instance.identifiers}
-                rowMetadata={['identifierTypeId']}
-                visibleColumns={['Resource identifier type', 'Resource identifier']}
-                formatter={identifiersRowFormatter}
-                ariaLabel="Identifiers"
-                containerRef={(ref) => { this.resultsList = ref; }}
-              />
+              <FormattedMessage id="ui-inventory.identifiers">
+                {ariaLabel => (
+                  <MultiColumnList
```
Comment (Godlevskyi) [inline src/ViewInstance.js]: Same here

File: src/ViewInstance.js [hunk 11]
```diff
@@ -570,18 +608,22 @@ class ViewInstance extends React.Component {
...
-                id="list-contributors"
-                contentData={instance.contributors}
-                visibleColumns={['Name type', 'Name', 'Type', 'Code', 'Source', 'Free text']}
-                formatter={contributorsRowFormatter}
-                ariaLabel="Contributors"
-                containerRef={(ref) => { this.resultsList = ref; }}
-              />
+              <FormattedMessage id="ui-inventory.contributors">
+                {ariaLabel => (
+                  <MultiColumnList
```
Comment (Godlevskyi) [inline src/ViewInstance.js]: Same here

File: src/ViewInstance.js [hunk 12]
```diff
@@ -590,18 +632,22 @@ class ViewInstance extends React.Component {
...
-                id="list-publication"
-                contentData={instance.publication}
-                visibleColumns={['Publisher', 'Publisher role', 'Place of publication', 'Publication date']}
-                formatter={publicationRowFormatter}
-                ariaLabel="Publication"
-                containerRef={(ref) => { this.resultsList = ref; }}
-              />
+              <FormattedMessage id="ui-inventory.publication">
+                {ariaLabel => (
+                  <MultiColumnList
```
Comment (Godlevskyi) [inline src/ViewInstance.js]: Same here

File: src/settings/ResourceTypesSettings.js [hunk 13]
```diff
@@ -1,36 +1,34 @@
...
   }
 
   suppressEdit = term => term.source === 'rdacontent';
   suppressDelete = term => term.source === 'rdacontent';
 
   render() {
-    const { formatMessage } = this.props.stripes.intl;
-
     return (
       <this.connectedControlledVocab
```
Comment (Godlevskyi) [inline src/settings/ResourceTypesSettings.js]: Please add columnMapping prop with the translations for the fields.
Please check all the occurrences for the connectedControlledVocab usage

File: src/Instances.js [hunk 14]
```diff
@@ -32,19 +35,19 @@ const languages = [
 // as those are pulled from the backend
 const filterConfig = [
   {
-    label: 'Resource Type',
+    label: <FormattedMessage id="ui-inventory.instances.resourceType" />,
```
Comment (Yurii-Danylenko) [inline src/Instances.js]: Please, check if there is no prop type warnings regarding to new type of the labels.

Comment (anastasiaHahanova) [reply src/Instances.js]: They are, so this PR is to resolve this:
https://github.com/folio-org/stripes-components/pull/789

File: src/ViewHoldingsRecord.js [hunk 15]
```diff
@@ -299,69 +338,88 @@ class ViewHoldingsRecord extends React.Component {
...
-                        'Statistical code type':
-                          x => this.refLookup(referenceTables.statisticalCodeTypes,
-                            this.refLookup(referenceTables.statisticalCodes, _.get(x, ['codeId'])).statisticalCodeTypeId).name,
-                        'Statistical code':
-                          x => this.refLookup(referenceTables.statisticalCodes, _.get(x, ['codeId'])).name,
-                      }}
-                      ariaLabel="Statistical codes"
-                      containerRef={(ref) => { this.resultsList = ref; }}
-                    />
+                    <FormattedMessage id="ui-inventory.statisticalCodes">
```
[the remark this answers is not in the document]
Comment (anastasiaHahanova) [reply src/ViewHoldingsRecord.js]: no, it won't work

[the remark this answers is not in the document]
Comment (cherewaty) [reply src/ViewHoldingsRecord.js]: Yup, this is probably the best we can do until @JohnC-80 has something new built with https://issues.folio.org/browse/STCOM-399

[the remark this answers is not in the document]
Comment (cherewaty) [reply src/ViewHoldingsRecord.js]: 🎉 https://github.com/folio-org/stripes-core/pull/535

[the remark this answers is not in the document]
Comment (anastasiaHahanova) [reply src/ViewHoldingsRecord.js]: Thanks @cherewaty, I've changed everything in accordance with STCOM-399 👍

**Pull request merged**